### PR TITLE
client: make Client.{Lstat,Open,Stat} satisfy os.IsNotExist

### LINKED ISSUE #71: (*Client).Stat and (*Client).Open don't return os.ErrNotExist for nonexistent paths
Running the following test:

    go test --integration

    func TestNonexistentPath(t *testing.T) {
            sftp, cmd := testClient(t, READONLY, NO_DELAY)
            defer cmd.Wait()
            defer sftp.Close()
    
            if _, err := sftp.Stat("/doesnt/exist/"); !os.IsNotExist(err) {
                    t.Errorf("os.IsNotExist(%v) = false, want true", err)
            }
    
            if _, err := sftp.Open("/doesnt/exist/"); !os.IsNotExist(err) {
                    t.Errorf("os.IsNotExist(%v) = false, want true", err)
            }
    }

Results in 

    --- FAIL: TestNonexistentPath (0.01s)
            client_integration_test.go:171: os.IsNotExist(sftp: "No such file" (SSH_FX_NO_SUCH_FILE)) = false, want true
            client_integration_test.go:175: os.IsNotExist(sftp: "No such file" (SSH_FX_NO_SUCH_FILE)) = false, want true
    process_write: write failed
    FAIL
    exit status 1
    FAIL        github.com/pkg/sftp     2.149s

### PR DESCRIPTION
**Pull request by mdlayher**
Fixes #71.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -1135,6 +1135,13 @@ func min(a, b int) int {
 	return a
 }
 
+func notExistOrErr(err error) error {
```
Comment (davecheney) [inline client.go]: what about somethign like 'normaliseError`

File: client.go [hunk 2]
```diff
@@ -1135,6 +1135,13 @@ func min(a, b int) int {
...
+	if err, ok := err.(*StatusError); ok && err.Code == ssh_FX_NO_SUCH_FILE {
```
Comment (davecheney) [inline client.go]: switch err := err.(type) {
case *StatusError {
     switch err.Code {
           case ssh_FX_...:
                   return os.ErrNotExist
           default:
                   return err
           }
default:
          return err
}

**Comment by davecheney** [comment]
lgtm with minor suggestions.

File: client.go (hunk 1 above)
Comment (mdlayher) [inline client.go]: That's fine.  I just followed the convention that was already in place in this section of the code.  Perhaps they could all be combined into one function?

Comment (mdlayher) [inline client.go]: See `okOrErr` and `eofOrErr`.

Comment (davecheney) [inline client.go]: @mdlayher I have a feeling that there will be more error codes to translate, so one function for all of them sounds like a good plan.

**Comment by mdlayher** [comment]
@davecheney Updated!  Please take a look.

**Comment by davecheney** [comment]
Very nice. Thanks.

**Pull request merged**